src/machine/machined-varlink.c

```diff
@@ -507,6 +508,112 @@ static int vl_method_terminate(sd_varlink *link, sd_json_variant *parameters, sd
+                        JSON_BUILD_PAIR_UNSIGNED_NON_ZERO("modificationTimestamp", image->mtime),
+                        JSON_BUILD_PAIR_UNSIGNED_NOT_EQUAL("usage", image->usage, UINT64_MAX),
+                        JSON_BUILD_PAIR_UNSIGNED_NOT_EQUAL("usageExclusive", image->usage_exclusive, UINT64_MAX),
+                        JSON_BUILD_PAIR_UNSIGNED_NOT_EQUAL("limit", image->limit, UINT64_MAX),
+                        JSON_BUILD_PAIR_UNSIGNED_NOT_EQUAL("limitExclusive", image->limit_exclusive, UINT64_MAX),
+                        SD_JSON_BUILD_PAIR_CONDITION(image->metadata_valid && !!image->hostname, "hostname", SD_JSON_BUILD_STRING(image->hostname)),
+                        SD_JSON_BUILD_PAIR_CONDITION(image->metadata_valid && !sd_id128_is_null(image->machine_id),
+                                                     "machineId",
+                                                     SD_JSON_BUILD_ID128(image->machine_id)),
+                        SD_JSON_BUILD_PAIR_CONDITION(image->metadata_valid && image->machine_info,
+                                                     "machineInfo",
+                                                     JSON_BUILD_STRV_ENV_PAIR(image->machine_info)),
+                        SD_JSON_BUILD_PAIR_CONDITION(image->metadata_valid && image->os_release,
+                                                     "OSRelease",
+                                                     JSON_BUILD_STRV_ENV_PAIR(image->os_release)));
+        if (r < 0)
+                return r;
+
+        if (more)
+                return sd_varlink_notify(link, v);
+
+        return sd_varlink_reply(link, v);
+}
+
+static int vl_method_list_images(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "name", SD_JSON_VARIANT_STRING, sd_json_dispatch_const_string, 0, 0 },
+                VARLINK_DISPATCH_POLKIT_FIELD,
+                {}
+        };
+
+        _cleanup_hashmap_free_ Hashmap *images = NULL;
+        const char *image_name = NULL;
+        Image *image;
+        int r;
+
+        assert(parameters);
+
+        r = sd_varlink_dispatch(link, parameters, dispatch_table, &image_name);
+        if (r != 0)
+                return r;
+
+        if (image_name) {
+                if (!image_name_is_valid(image_name))
+                        return sd_varlink_error_invalid_parameter_name(link, "name");
+
+                r = image_find(IMAGE_MACHINE, image_name, /* root = */ NULL, &image);
+                if (r == -ENOENT)
+                        return sd_varlink_error(link, "io.systemd.Machine.NoSuchImage", NULL);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to find image: %m");
+
+                // TODO(ikruglov): decide if we want to load metadata in the general case of listing
```

Comment:
I'd really make this an option. i.e. add a boolean argument called acquireMetadata or so, which defaults to false

Reply:
not sure i grok your question? just make it a boolean field passed to the call which we apply to all machines we enumerate